modules/cnvkit/1.0/cnvkit.smk

```diff
@@ -0,0 +1,318 @@
+                        seq_type = CFG["runs"]["tumour_seq_type"]),
+        controls = get_normals,
+        refFlat = CFG["dirs"]["inputs"] + "ref/refFlat.txt",
+        fasta =  reference_files("genomes/{genome_build}{masked}/genome_fasta/genome.fa"),
+    output: 
+        panel_of_norms = CFG["dirs"]["cnvkit"] + "{seq_type}--{genome_build}{masked}/{capture_space}/normal_reference.cnn",
+        cnr = expand(
+                CFG["dirs"]["cnvkit"] + "{{seq_type}}--{{genome_build}}{{masked}}/{{capture_space}}/{tumour_id}.cnr", zip,
+                tumour_id = CFG["runs"]["tumour_sample_id"]),
+        cns = expand(
+                CFG["dirs"]["cnvkit"] + "{{seq_type}}--{{genome_build}}{{masked}}/{{capture_space}}/{tumour_id}.cns", zip,
+                tumour_id = CFG["runs"]["tumour_sample_id"]),
+        call_cns = expand(
+                CFG["dirs"]["cnvkit"] + "{{seq_type}}--{{genome_build}}{{masked}}/{{capture_space}}/{tumour_id}.call.cns", zip,
+                tumour_id = CFG["runs"]["tumour_sample_id"]),
+    params:
+        outdir = CFG["dirs"]["cnvkit"] + "{seq_type}--{genome_build}{masked}/{capture_space}/",
+        targets = op.switch_on_wildcard("capture_space", CFG["options"]["target_bed"]),
+        opts = CFG["options"]["batch"]
+    conda:
+        CFG["conda_envs"]["cnvkit"]
+    threads:
+        CFG["threads"]["batch"]
+    resources:
+        **CFG["resources"]["batch"]
+    log: 
+        stdout = CFG["logs"]["cnvkit"] + "{seq_type}--{genome_build}{masked}/{capture_space}_cnr.log"
+    shell:
+        "cnvkit.py batch {input.samples} --normal {input.controls} --fasta {input.fasta} --annotate {input.refFlat} --processes {threads} --output-reference {output.panel_of_norms} --targets {params.targets} --output-dir {params.outdir} {params.opts} > {log} 2&>1 "
+
+
+rule _cnvkit_symlink_cns:
+    input:
+        call_cns = CFG["dirs"]["cnvkit"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.call.cns"
+    output:
+        call_cns = CFG["dirs"]["cn_calls"] + "{seq_type}--{genome_build}{masked}/{capture_space}/{tumour_id}.call.cns"
+    run:
+        op.relative_symlink(input.call_cns, output.call_cns, in_module = True)
+
+
+# need SNPs not SNVs (i.e. get germline calls using a dbSNP vcf)
+rule _cnvkit_dbsnp_to_bed:
+    input:
+        vcf = reference_files("genomes/{genome_build}/variation/dbsnp.common_all-151.vcf.gz")
+    output:
+        bed = CFG["dirs"]["SNPs"] + "{genome_build}/dbsnp.common_all-151.bed"
+    resources: **CFG["resources"]["SNPs"]
+    shell:
+        op.as_one_line(""" gunzip -c {input.vcf} | awk {{'printf ("%s\\t%s\\t%s\\n", $1,$2-1,$2)'}} | zgrep -v -h "^#" > {output.bed} """)
+
+#### set-up mpileups for BAF calling ####
+def _cnvkit_get_chr_mpileups(wildcards):
+    CFG = config["lcr-modules"]["cnvkit"]
+    chrs = reference_files("genomes/" + wildcards.genome_build + "/genome_fasta/main_chromosomes_withY.txt")
```

Comment:
This is a good catch. CNVkit can apparently determine sex just based on the coverage on the X and Y in the sample/normal. There is a flag -y to assume a male reference and (without -y , the default) would be for a female reference. The only difference with the flag is just the adjustments to the chrX and chrY copy numbers - without -y flag, a male would have log.ratio -1 for chrX and 0 for chr Y, while a female would have 0 for chrX and -Inf for chrY.

I think since cnvkit has that as an output already, we may as well keep it in. If we need to keep it consistent with other CNV modules, we can just simply remove all chrY calls in the final seg file.